src/Library/demos/Carousel/main.py

```diff
@@ -0,0 +1,67 @@
+import gi
+
+gi.require_version("Gtk", "4.0")
+gi.require_version("Adw", "1")
+
+from gi.repository import Adw, Gtk
+import workbench
+
+root_box = workbench.builder.get_object("root_box")
+carousel = workbench.builder.get_object("carousel")
+ls_switch = workbench.builder.get_object("ls_switch")
+sw_switch = workbench.builder.get_object("sw_switch")
+indicator_row = workbench.builder.get_object("indicator_row")
+orientation_row = workbench.builder.get_object("orientation_row")
+indicators = None
+
+carousel.connect("page-changed", lambda *_: print("Page Changed"))
+
+# Scroll Wheel Switch
+sw_switch.set_active(carousel.get_allow_scroll_wheel())
+
+sw_switch.connect(
+    "notify::active", lambda *_: carousel.set_allow_scroll_wheel(sw_switch.get_active())
+)
+
+# Long Swipe Switch
+ls_switch.set_active(carousel.get_allow_long_swipes())
+
+ls_switch.connect(
+    "notify::active", lambda *_: carousel.set_allow_long_swipes(ls_switch.get_active())
+)
+
+if indicator_row.get_selected() == 0:
+    indicators = Adw.CarouselIndicatorDots(carousel=carousel)
+else:
+    indicators = Adw.CarouselIndicatorLines(carousel=carousel)
+
+indicators.set_orientation(carousel.get_orientation())
+root_box.append(indicators)
+
+
+def on_indicator_selected(_widget, _item):
+    global indicators
```

Comment:
This is OK in this case, you need to do this since you are below declaring a local variable with the same name as the global one. If you did not have a local variable with the same name, Python would automatically use the global. 
But since you do set a variable "indicators" below Python assumes all usage of that variable to be local, giving you an error about undefined locals on the next line. By declaring indicators as global you explicitly tell Python that your setting of indicators is not declaring a local variable, it is modifying the global one and all usages of indicators are again referring to that global variable.

In real Python code you'd probably avoid doing this to avoid confusion, but since this is a small demo and we want to keep these close to the JavaScript demos this is OK. Personally I'd probably use something like a partial to pass the indicators into the function.

Reply:
>  Personally I'd probably use something like a partial to pass the indicators into the function.

So do I keep it as is, or do this?

Reply:
I would keep it like is for now to stay closer to the JS demo.

Reply:
Then I think this can be merged.